src/pykeen/triples/triples_factory.py

```diff
@@ -162,6 +168,304 @@ def _map_triples_elements_to_ids(
+    unknown_label: str,
+) -> np.ndarray:
+    """Apply vectorized labeler to an array of IDs.
+
+    :param mapped_column:
+        The IDs.
+    :param vectorized_labeler:
+        The vectorized labeler.
+    :param unknown_label:
+        A label to use for unknown IDs.
+
+    :return:
+        An array of labels, same shape as mapped_column.
+    """
+    # Input normalization
+    if isinstance(mapped_column, int):
+        mapped_column = [mapped_column]
+    if torch.is_tensor(mapped_column):
+        mapped_column = mapped_column.cpu().numpy()
+    mapped_column = np.asanyarray(mapped_column)
+
+    # Actual labeling
+    return vectorized_labeler(mapped_column, (unknown_label,))
+
+
+@dataclass
+class LabelMapping:
+    """
+    A mapping between labels to IDs.
+
+    Within this class we use the following terminology:
+
+    * mapping:
+        A conversion from labels to IDs.
+    * labeling:
+        A conversion from IDs to labels.
+    """
+
+    #: The mapping for entities
+    entity_label_to_id: EntityMapping
+
+    #: The mapping for relations
+    relation_label_to_id: RelationMapping
+
+    #: A dictionary mapping each relation to its inverse, if inverse triples were created
+    relation_to_inverse: Optional[Mapping[str, str]] = None
+
+    #: The inverse mapping for entity_label_to_id; initialized automatically
+    entity_id_to_label: Mapping[int, str] = None
+
+    #: The inverse mapping for relation_label_to_id; initialized automatically
+    relation_id_to_label: Mapping[int, str] = None
+
+    #: A vectorized version of entity_label_to_id; initialized automatically
+    _vectorized_entity_mapper: Callable[[np.ndarray, Tuple[int]], np.ndarray] = None
+
+    #: A vectorized version of relation_label_to_id; initialized automatically
+    _vectorized_relation_mapper: Callable[[np.ndarray, Tuple[int]], np.ndarray] = None
+
+    #: A vectorized version of entity_id_to_label; initialized automatically
+    _vectorized_entity_labeler: Callable[[np.ndarray, Tuple[str]], np.ndarray] = None
+
+    #: A vectorized version of relation_id_to_label; initialized automatically
+    _vectorized_relation_labeler: Callable[[np.ndarray, Tuple[str]], np.ndarray] = None
+
+    @staticmethod
+    def from_labeled_triples(triples: LabeledTriples) -> 'LabelMapping':
+        """Create a mapping from labeled triples."""
+        return LabelMapping(
+            entity_label_to_id=create_entity_mapping(triples=triples),
+            relation_label_to_id=create_relation_mapping(relations=triples[:, 1])
+        )
+
+    def __post_init__(self):
+        """Pre-compute derived mappings."""
+        # ID to label mapping
+        self.entity_id_to_label = invert_mapping(mapping=self.entity_label_to_id)
+        self.relation_id_to_label = invert_mapping(mapping=self.relation_label_to_id)
+
+        # vectorized versions
+        self._vectorized_entity_mapper = np.vectorize(self.entity_label_to_id.get)
```

Comment:
Using ``self._vectorized_entity_mapper = np.vectorize(self.entity_label_to_id.__getitem__)`` might be faster since it doesn't check for it being there or not and raises an error on missing rather than returning none. Same for the others

Reply:
oh but later it looks like you're giving support for *wrong* entries

Reply:
For mapper we support this in order to filter out triples with unknown entities rather than just crashing. So, for consistency, I also support labeling with an "unknown" entity label.